swarm/storage/chunker_test.go

```diff
@@ -40,136 +39,45 @@ type test interface {
-					// self.chunks = append(self.chunks, chunk)
-					self.chunks[chunk.Key.Hex()] = chunk
-					close(chunk.dbStored)
-				}
-
-			}
-		}()
-	}
+func (t *testChunkStore) Put(chunk *Chunk) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.chunks[chunk.Key.Hex()] = chunk
+	close(chunk.dbStored)
+}
 
-	var w func()
-	key, w, err = chunker.Split(data, size, chunkC)
-	if err != nil && expectedError == nil {
-		err = fmt.Errorf("Split error: %v", err)
-	}
-	if chunkC != nil {
-		wait = func() {
-			w()
-			close(quitC)
-		}
-	} else {
-		wait = func() {}
+func (t *testChunkStore) Get(key Key) (*Chunk, error) {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+	chunk := t.chunks[key.Hex()]
+	if chunk == nil {
+		return nil, ErrChunkNotFound
 	}
-	return key, wait, err
+	return chunk, nil
 }
 
-func (self *chunkerTester) Append(chunker Splitter, rootKey Key, data io.Reader, chunkC chan *Chunk, expectedError error) (key Key, wait func(), err error) {
-	quitC := make(chan bool)
-	timeout := time.After(60 * time.Second)
-	if chunkC != nil {
-		go func() error {
-			for {
-				select {
-				case <-timeout:
-					return errors.New("Append timeout error")
-				case <-quitC:
-					return nil
-				case chunk := <-chunkC:
-					if chunk != nil {
-						stored, success := self.chunks[chunk.Key.Hex()]
-						if !success {
-							// Requesting data
-							self.chunks[chunk.Key.Hex()] = chunk
-							close(chunk.dbStored)
-						} else {
-							// getting data
-							chunk.SData = stored.SData
-							chunk.Size = int64(binary.LittleEndian.Uint64(chunk.SData[0:8]))
-							close(chunk.dbStored)
-							if chunk.C != nil {
-								close(chunk.C)
-							}
-						}
-					}
-				}
-			}
-		}()
-	}
-	var w func()
-	key, w, err = chunker.Append(rootKey, data, chunkC)
-	if err != nil && expectedError == nil {
-		err = fmt.Errorf("Append error: %v", err)
-	}
+func (t *testChunkStore) Close() {
+}
 
-	if chunkC != nil {
-		wait = func() {
-			w()
-			close(quitC)
-		}
-	} else {
-		wait = func() {}
+func newTestChunkStore() *testChunkStore {
+	return &testChunkStore{
+		chunks: make(map[string]*Chunk),
 	}
-	return key, wait, err
 }
 
-func (self *chunkerTester) Join(chunker Chunker, key Key, c int, chunkC chan *Chunk, quitC chan bool) LazySectionReader {
-	// reset but not the chunks
-
-	timeout := time.After(600 * time.Second)
-	i := 0
-	go func() error {
-		for {
-			select {
-			case <-timeout:
-				return errors.New("Join timeout error")
-			case chunk, ok := <-chunkC:
-				if !ok {
-					close(quitC)
-					return nil
-				}
-				// this just mocks the behaviour of a chunk store retrieval
-				stored, success := self.chunks[chunk.Key.Hex()]
-				if !success {
-					return errors.New("Not found")
-				}
-				chunk.SData = stored.SData
-				chunk.Size = int64(binary.LittleEndian.Uint64(chunk.SData[0:8]))
-				close(chunk.C)
-				i++
-			}
-		}
-	}()
-
-	reader := chunker.Join(key, chunkC, 0)
-	return reader
+func newTestHasherStore(chunkStore ChunkStore, hash string) *hasherStore {
```

Comment:
Nitpick, but I think this method is redundant - just use directly the `NewHasherStore`.